Makefile

```diff
@@ -1,17 +1,33 @@
+.PHONY: all fmt check-fmt lint lint-fix codespell all
+
+DESTDIR ?=
 
 ALL_SRC_FILES = 'find . -name "*.[ch]" ! -path "./src/libhirte/ini/ini.[ch]" ! -path "./src/libhirte/hashmap/*" ! -path "./src/*/test/**" ! -path "./builddir/**" -print0'
 
+all:
+	meson setup builddir
+	meson compile -C builddir
+
+test: all
+	meson setup builddir
```

Comment:
`setup` and `compile` can be removed from the `test` target since it is calling `all`.